src/Sarif/TypedPropertiesDictionary.cs

```diff
@@ -13,7 +13,7 @@ public interface IMarker { }
 
     [Serializable]
     [JsonConverter(typeof(TypedPropertiesDictionaryConverter))]
-    public class TypedPropertiesDictionary<T> : Dictionary<string, T>, IMarker where T : new()
+    public class TypedPropertiesDictionary<T> : ConcurrentDictionary<string, T>, IMarker where T : new()
```

Comment:
>ConcurrentDictionary [](http://example.com/codeflow?start=48&length=20)

Update the title of the PR and add a new entry to the ReleaseHistory.

The title should be something like: "Fix InvalidOperationException when using PropertiesDictionary in a multithreaded application".

With that, if any customer checks the release history, the customer would know what motivated us to change this from Dictionary to ConcurrentDictionary.

This should also be applied to the PR description.

Reply:
done both, thanks

Reply:
also added the ReleaseHistory for the other 2 PRs.